File: back/back/Models/DroitGroupe.cs
using LiteDB;

namespace back.Models;

/// <summary>
///  Parametrer les droits des comptes sur l'application
/// </summary>
public sealed class DroitGroupe
{
     [BsonId]
     public int Id { get; set; }

     public string Nom { get; set; } = null!;

     public bool PeutAcheterVaisseau { get; set; }
     public bool PeutAcheterLogistiqueMateriel { get; set; }

     /// <summary>
     /// Droit pour chaque Route du groupe de route
     /// </summary>
     public List<Droit> ListeDroit { get; set; } = [];
}

public sealed class Droit
{
     public string RouteGroupe { get; set; } = null!;
     public bool PeutLire { get; set; }
     public bool PeutEcrire { get; set; }
     public bool PeutSupprimer { get; set; }
}

File: front/src/models/DroitGroupe.ts
export type DroitGroupe = 
{
    id: number,
    nom: string,
    peutProposerLogistiqueMateriel: boolean,
    peutAcheterLogistiqueMateriel: boolean,
    peutAcheterVaisseau: boolean,
    listeDroit: Droit[]
}

export type Droit =
{
    routeGroupe: string,
    peutLire: boolean,
    peutEcrire: boolean,
    peutSupprimer: boolean
}

export type DroitGroupeRequete = Omit<DroitGroupe, "id">;